src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -408,6 +420,7 @@ private RequestObjectParamView generateRequestObjectParam(
     SurfaceNamer namer = context.getNamer();
     RequestObjectParamView.Builder param = RequestObjectParamView.newBuilder();
     param.name(namer.getVariableName(field));
+    param.nameAsMethodName(namer.methodName(Name.lowerCamel(namer.getVariableName(field))));
```

Comment:
Thanks for the explanation. This is fixed by moving the casing logic to SurfaceNamer, as you suggest.